### PR TITLE
Output web test artifacts to out directory.

### PR DESCRIPTION
**Pull request by eyebrowsoffire**

### CODE REVIEW

File: lib/web_ui/dev/environment.dart [hunk 1]
```diff
@@ -150,8 +150,8 @@ class Environment {
   ///
   /// This is where compiled output goes.
   io.Directory get webUiBuildDir => io.Directory(pathlib.join(
-        webUiRootDir.path,
-        'build',
+        outDir.path,
```
Comment (godofredoc) [inline lib/web_ui/dev/environment.dart:153]: Can we copy to both locations as a transitional step to not break the current tests?

File: lib/web_ui/dev/licenses.dart [hunk 2]
```diff
@@ -79,13 +79,18 @@ class LicensesCommand extends Command<bool> {
   }
 
   List<io.File> _flatListSourceFiles(io.Directory directory) {
+    // This is the old path that tests used to be built into. Ignore anything
+    // within this path.
+    final String legacyBuildPath = path.join(environment.webUiRootDir.path, 'build');
```
Comment (mdebbar) [inline lib/web_ui/dev/licenses.dart:84]: Should we also add the legacy path to the `felt clean` command?
https://github.com/flutter/engine/blob/b2720b7e9887ff0c000d43f5323c9b4995df7831/lib/web_ui/dev/clean.dart#L40-L49

I personally rarely use `felt clean` so I don't have a strong opinion here.

File: lib/web_ui/dev/environment.dart (hunk 1 above)
Comment (eyebrowsoffire) [reply lib/web_ui/dev/environment.dart:153]: I don't think we'll have to do that. AFAIK the current tests will not be broken by this.

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/dev/licenses.dart (hunk 2 above)
Comment (eyebrowsoffire) [reply lib/web_ui/dev/licenses.dart:84]: Yeah good idea

**Comment by Hixie** [comment]
test-exempt: is tests

**Pull request merged**